File: src/data/courses/dsa/dynamic-programming/linear-1d-dp/house-robber/code/python/house_robber.py
class Solution:
    def rob(self, nums) -> int:
        if len(nums) == 1: return nums[0]
        a, b = nums[0], max(nums[0], nums[1])
        for i in range(2, len(nums)): a, b = b, max(b, a + nums[i])
        return b

File: src/data/courses/dsa/dynamic-programming/linear-1d-dp/house-robber/code/java/HouseRobber.java
class Solution {
    public int rob(int[] nums) {
        if (nums.length == 1) return nums[0];
        int a = nums[0], b = Math.max(nums[0], nums[1]);
        for (int i = 2; i < nums.length; i++) { int c = Math.max(b, a+nums[i]); a=b; b=c; }
        return b;
    }
}